### PR TITLE
Maint/improve coverage

### PR DESCRIPTION
**Pull request by mfeurer**
First split of #150

### CODE REVIEW

**Comment by mfeurer** [comment]
@amueller ready to be reviewed/merged

File: openml/runs/run.py [hunk 1]
```diff
@@ -47,8 +47,8 @@ def __init__(self, task_id, flow_id, setup_string, dataset_id, files=None,
         self.data_content = data_content
         self.model = model
 
-    def _generate_arff(self):
-        """Generates an arff for upload to server.
+    def _generate_arff_header_dict(self):
```
Comment (amueller) [inline openml/runs/run.py]: I think the docstring for the return value also needs to be changed

File: openml/runs/run.py [hunk 2]
```diff
@@ -167,26 +177,21 @@ def run_task(task, model):
...
             PredY = model.predict(testX)
-            end_time = time.time()
-
-            train_times.append(end_time - start_time)
 
             for i in range(0, len(test_indices)):
-                arff_line = [rep_no, fold_no, test_indices[i],
-                             class_labels[PredY[i]], class_labels[testY[i]]]
-                arff_line[3:3] = ProbaY[i]
+                arff_line = [rep_no, fold_no, test_indices[i]]
```
Comment (amueller) [inline openml/runs/run.py]:
```python
arff_lline = [repo_no, fold_no, test_indices[i]] + ProbaY[i] + [class_labels[PredY[i]], class_labels[testY[i]]]
```
or if too long
```python
arff_lline = ([repo_no, fold_no, test_indices[i]] + ProbaY[i]
                  + [class_labels[PredY[i]], class_labels[testY[i]]])
```

**Comment by amueller** [comment]
looks good apart from minor comments

**Comment by mfeurer** [comment]
Okay, I will add the comments, fix the conflicts and merge it tomorrow morning.

File: openml/runs/run.py (hunk 2 above)
Comment (mfeurer) [inline openml/runs/run.py]: This returns an error because ProbaY is a numpy array which can't be concatenated like this. Is this a problem?

**Comment by mfeurer** [comment]
Okay, I decided to try to finish this today. Can I merge this?

File: openml/runs/run.py (hunk 2 above)
Comment (amueller) [inline openml/runs/run.py]: not really ``tolist()`` would be a fix but not really important

**Comment by amueller** [comment]
feel free to merge.

**Pull request merged**